### ISSUE TITLE
Use Shellcheck to resolve shell script violations

### ISSUE THREAD

**Issue by Rahulkhinchi03** [checkstyle/checkstyle#11637]
ShellCheck is an open-source static analysis tool that automatically finds bugs in your shell scripts. (https://github.com/koalaman/shellcheck)

`./.ci/*sh` has all the shell scripts files and we have disabled all the violations at the `.shellcheckrc` file.
We are currently invoking spellcheck through Terminal (You can download it through apt or similar package managers)

TO-DO:
Pick a suppression (single suppression  fix in single Pull Request) To allocate a suppression update for you please make comment in the issue "I am on XxxxxSuppression"
Remove Your Selected Suppression from the `.shellcheckrc` file.
Run shellcheck through Terminal and See Errors
Fix the Errors and make a PR.

LIST OF SUPPRESSIONS:

- [ ] SC2086 # (info): Double quote to prevent globbing and word splitting.
- [ ] SC2006 # (style): Use $(...) notation instead of legacy backticks ....
- [ ] SC2046 # (warning): Quote this to prevent word splitting.
- [ ] SC2207 # (warning): Prefer mapfile or read -a to split command output.
- [ ] SC2002 # (style): Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
- [ ] SC2061 # (warning): Quote the parameter to -name so the shell won't interpret it.
- [ ] SC2035 # (info): Use ./glob or -- glob so names with dashes won't become options.
- [ ] SC2236 # (style): Use -n instead of ! -z.
- [ ] SC2016 # (info): Expressions don't expand in single quotes, use double quotes for that.
- [ ] SC2185 # (info): Some finds don't have a default path. Specify '.' explicitly.
- [ ] SC2155 # (warning): Declare and assign separately to avoid masking return values.
- [ ] SC2071 # (error): > is for string comparisons. Use -gt instead.
- [ ] SC2034 # (warning): RUN_JOB appears unused. Verify use (or export if used externally).
- [ ] SC2216 # (warning): Piping to 'true', a command that doesn't read stdin.
- [ ] SC2013 # (info): To read lines rather than words, pipe/redirect to a 'while read' loop.
- [ ] SC2206 # (warning): Quote to prevent word splitting/globbing.
- [ ] SC2143 # (style): Use grep -q instead of comparing output with [ -n .. ].
- [ ] SC2004 # (style): $/${} is unnecessary on arithmetic variables.
- [ ] SC2087 # (warning): Quote 'EOF' to make here document expansions happen on the server-side.
- [ ] SC2102 # (info): Ranges can only match single chars (mentioned due to duplicates).
- [ ] SC2242 # (error): Can only exit with status 0-255.
- [ ] SC2115 # (warning): Use "${var:?}" to ensure this never expands to /* .
- [ ] SC2128 # (warning): Expanding an array without an index only gives the first element.
- [ ] SC2063 # (warning): Grep uses regex, but this looks like a glob.
- [ ] SC2126 # (style): Consider using grep -c instead of grep|wc -l.
- [ ] SC2088 # (warning): Tilde does not expand in quotes. Use $HOME.
- [ ] SC2005 # (style): Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
- [ ] SC2015 # (info): Note that A && B || C is not if-then-else. C may run when A is true.
- [ ] SC2026 # (info): This word is outside of quotes.
- [ ] SC2269 # (info): This variable is assigned to itself, so the assignment does nothing.

**Comment by Rahulkhinchi03** [comment]
@romani @rnveach @nick-mancuso The `wercker.sh` violations are excluded in the issue as https://github.com/checkstyle/checkstyle/issues/11564

**Comment by Rahulkhinchi03** [comment]
@romani @nick-mancuso @rnveach 
Before the Project gets started, Should I resolve all the shell code errors in the `release.sh` file?

**Comment by romani** [comment]
Yes, but Better to slice a work by rules to let us activate rule and make sure nobody will introduce problems again.

**Comment by Rahulkhinchi03** [comment]
Sure, I can see that `release.sh`  has error codes: `SC2046`, `SC2086`, `SC2006`, `SC2035`.

**Comment by stoyanK7** [comment]
I am on `SC2269` and `SC2046`

**Comment by Rahulkhinchi03** [comment]
@stoyanK7 That's really great!
Please make different PRs for both the ERROR CODES.

**Comment by shubh220922** [comment]
I'm on ``SC2005``

**Comment by Kevin222004** [comment]
I am on  SC2088

**Comment by shubh220922** [comment]
I am on ``SC2126``

**Comment by shubh220922** [comment]
@romani 

Kindly update the checklist. Shellcheck ``SC2006`` is already resolved but, not checked in the checklist(https://github.com/checkstyle/checkstyle/issues/11637#:~:text=SC2006%20%23%20(style)%3A%20Use%20%24(...)%20notation%20instead%20of%20legacy%20backticks%20....)

**Comment by shubh220922** [comment]
I'm on ``SC2002``, ``SC2016``, ``SC2061``.
